R/set-record-option.R

```diff
@@ -0,0 +1,110 @@
+
+#' Add or modify a record option
+#'
+#' Add a new record option or adjust the value/presence of an existing one.
+#'
+#' Note that only estimation, data, and table records are currently supported.
+#'
+#' Notes on `value`:
+#'  - We infer whether to use `option_value` or `option_flag` for *new* options based on the *class* of `value`:
+#'     - If you are trying to add an option with an assigned value (i.e. `MAXEVAL = 10`), `value` must be a character string or numeric value
```

Comment:
It's super minor and a matter of taste, but for consistency with other docs, I'd appreciated if you kept the max line length to around 80 characters.